### DISCUSSION TITLE
浅色背景下默认白色弹幕看不清，是否有可能实现一个修改弹幕颜色/渲染方式的功能？

### DISCUSSION THREAD

**Discussion by Adiwon** [the1812/Bilibili-Evolved#4407] [功能建议 / Ideas]
B站19年更新播放器后就一直有这个问题：在浅色的视频背景上，白色弹幕根本看不清。现有的三种描边类型（秒变、重墨、45度投影）换来换去也没啥用。有没有可能让默认弹幕显示为灰色/黑色之类的颜色，甚至直接加个半透明背景框框，从而提高可见性？

有个例子，19年之前的旧版播放器，默认的css3渲染方式也有这种问题，但改成Canvas之后弹幕可见性就高多了，不知是否可以作为参考。

当前版本默认效果：
<img width="176" alt="001" src="https://github.com/the1812/Bilibili-Evolved/assets/16150302/b3242d78-109f-4804-af88-808322fc053f">

旧版 CSS3：
<img width="251" alt="002" src="https://github.com/the1812/Bilibili-Evolved/assets/16150302/8876317b-666a-4a34-b2a4-f0ee7538e524">

旧版 Canvas：
<img width="249" alt="003" src="https://github.com/the1812/Bilibili-Evolved/assets/16150302/9b7dd683-e196-47a6-b0eb-b15d403d71cf">

**Comment by the1812** [comment]
你浏览器有问题吧, 重墨, 描边, 投影顶多描边有点难看清
![Snipaste_2023-09-05_21-06-33](https://github.com/the1812/Bilibili-Evolved/assets/26504152/5d35ebf5-ccb5-40f0-9427-d55942488d8d)
![Snipaste_2023-09-05_21-06-56](https://github.com/the1812/Bilibili-Evolved/assets/26504152/d6146bca-9c3f-4109-a97a-2427600e1d65)
![Snipaste_2023-09-05_21-07-22](https://github.com/the1812/Bilibili-Evolved/assets/26504152/1ed1d854-cc23-493a-91b8-c71d92bf8533)

**Reply by Adiwon** [reply]
排查出来了……我一直在用stylus里的“edge chrome 字体渲染美化”这个样式，只要启用，弹幕就会看不清……
